proto/cosmos/crypto/ledger/keys.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+package cosmos.crypto.ledger;
+
+import "cosmos/crypto/secp256k1/keys.proto";
+import "cosmos/crypto/hd/v1beta1/hdpath.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/ledger";
+
+// PrivKeyLedgerSecp256K1 implements PrivKey, calling the ledger nano we
+// cache the PubKey from the first call to use it later.
+message PrivKeyLedgerSecp256K1 {
+  // CachedPubKey should be private, but we want to encode it via
+  // go-amino so we can view the address later, even without having the
+  // ledger attached.
+  cosmos.crypto.secp256k1.PubKey cached_pub_key = 1;
```

Comment:
This seems to me like an implementation detail, and should not be exposed in the proto file.

Could we create our own:

```go 
type OurOwnPrivKeyLedgerSecp256K1 struct {
  pb.PrivKeyLedgerSecp256K1
  CachedPubKey
}
```

Reply:
Actually, do we need `PrivKeyLedgerSecp256k1` to be a proto.message?

We could just not let it implement `cryptotypes.PrivKey`, but still keep the `.Sign()`, `.PubKey()` methods.

Reply:
why not keeping the other `cryptotypes.PrivKey` methods?
I also have tried the route of having `PrivKeyLedgerSecp256k1` just implement the regular `PrivKey` methods interface (without `proto.Message`) and this was pretty messy.

Reply:
> and this was pretty messy

Got it 👍. If there's a way to not put the `cached_pub_key` in the proto file, then I'm good.

Reply:
Well, that made me think it through a little bit more and actually they could be a cleaner way to do that using some `BasePrivKey` interface (`PrivKey` without `proto.Message`) so I'll try to re-explore that route!
Otherwise if we have to with proto type, agreed to remove the `cached_pub_key` field, thanks!